service/qos/service_level_controller.cc

```diff
@@ -341,31 +340,33 @@ future<> service_level_controller::alter_distributed_service_level(sstring name,
 }
 
 future<> service_level_controller::drop_distributed_service_level(sstring name, bool if_exists) {
-    co_await _sl_data_accessor->start_service_level_operation();
-    auto sl_info = co_await _sl_data_accessor->get_service_levels();
+    return maybe_execute_on_shard0([name, if_exists] (service_level_controller& sl_controller) -> future<> {
+        co_await sl_controller._sl_data_accessor->start_service_level_operation();
+        auto sl_info = co_await sl_controller._sl_data_accessor->get_service_levels();
 
-    auto it = sl_info.find(name);
-    if (it == sl_info.end()) {
-        _sl_data_accessor->abort_service_level_operation();
```

Comment:
Is this all just a big indent to fit in the execute_on_shard_0? Better to use a half-indent (two spaces) for maybe_execute_on_shard0() so it's readable. Or alternatively rename this to drop_distributed_service_level_on_shard0() and add a new drop_distributed_service_level() that calls it.

Reply:
now whole cql-statement is executed on shard0, so there is no need for `maybe_execute_on_shard0()`